plugins/clipboard/plugin.js

```diff
@@ -114,7 +114,36 @@
 'use strict';
 
 ( function() {
-	var clipboardIdDataType;
+	var clipboardIdDataType,
+		jpeg2000Extensions = [ 'jp2', 'j2k', 'jpf', 'jpm', 'jpg2', 'j2c', 'jpc', 'jpx' ],
+		heifExtensions = [ 'heif', 'heifs', 'heic', 'avci', 'avcs', 'avif', 'avifs' ],
+		jpegXRExtensions = [ 'jxr', 'hdp', 'wdp' ],
+		windowsBitmapExtension = [ 'bmp', 'dib' ],
+		tiffExtensions = [ 'tif', 'tiff' ],
+		xBitmapExtension = [ 'xbm' ],
+		imageMimeTypes = {
+			'image/jpeg': [ 'jpg', 'jpeg' ,'jpe', 'jif', 'jfi', 'jfif', 'pjpeg', 'pjp' ],
+			'image/x-icon': [ 'ico', 'cur' ],
+			'image/apng': [ 'apng' ],
+			'image/webp': [ 'webp' ],
+			'image/png': [ 'png' ],
+			'image/gif': [ 'gif' ],
+			'image/svg': [ 'svg' ],
+			'image/vnd.ms-photo': jpegXRExtensions,
+			'image/bmp': windowsBitmapExtension,
+			'image/x-bmp': windowsBitmapExtension,
+			'image/jpm': jpeg2000Extensions,
+			'image/jpx': jpeg2000Extensions,
+			'image/jp2': jpeg2000Extensions,
+			'image/xbm': xBitmapExtension,
+			'image/xbitmap': xBitmapExtension,
+			'image/jxr': jpegXRExtensions,
+			'image/tiff-fx': tiffExtensions,
+			'image/tiff': tiffExtensions,
+			'image/avif': heifExtensions,
+			'image/heif': heifExtensions,
+			'image/heic': heifExtensions
+		};
```

Comment:
I'm wondering if switching solely to file extensions wouldn't be simpler – we wouldn't need to keep such a map up to date.

The only real downside I could see is files without extensions or files with wrong extensions (e.g. text file inside `.gif` file).